src/com/facebook/buck/android/DxStep.java

```diff
@@ -162,9 +163,42 @@ public DxStep(
       Optional<String> maxHeapSize,
       String dexTool,
       boolean intermediate) {
+    this(
+        filesystem,
+        androidPlatformTarget,
+        outputDexFile,
+        filesToDex,
+        options,
+        maxHeapSize,
+        dexTool,
+        intermediate,
+        null);
+  }
+
+  /**
+   * @param outputDexFile path to the file where the generated classes.dex should go.
```

Comment:
D8 can generate multiple dex files in a single step. Can we fix the assumption that buck makes currently which is always expecting a single dex as output from D8?

Reply:
Is it necessary for this PR? We can probably do that in a separate PR. Moreover I've live to experiment with dex merge using pure D8 and allow it producing primary and secondary dex files as it designed. Not using current split mechanism. Again that's a separate change.

Reply:
That makes sense. Can be done separately, but it will be a big win to do so later since buck currently computes the method and field counts via an asm pass after compilation so it can play with the rules of the old dexer. Since that is not needed anymore, we can avoid all that overhead with D8